### DISCUSSION TITLE
How to hide enviromental variables in browser?

### DISCUSSION THREAD

**Discussion by coolpi21** [remix-run/remix#8886] [Q&A]
Once I finished deploying my Remix app, it became apparent that data was being pulled from the loader function and fed into the __remixContext object. This included certain private environmental variables of mine. But now there are a few particularly sensitive environmental variables that I'd prefer to keep hidden. what's the best way for me to go about doing so?

**Comment by sergiodxa** [comment]
Any data you return from a loader or action is going to be sent to the browser

If you have an env variable that needs to remain private don’t send it in the loader response.

If you need the value in the UI then it needs to be public anyway.
